docs/spec/staking/end_block.md

```diff
@@ -1,68 +1,40 @@
-## Unbonding Validator Queue
+## Validator Set Changes
 
-For all unbonding validators that have finished their unbonding period, this switches their validator.Status
-from sdk.Unbonding to sdk.Unbonded if they still have any delegation left.  Otherwise, it deletes it from state.
+The Tendermint validator set is be updated by state transitions that run at the
+end of every block. Operations are as following:
 
-```golang
-validatorQueue(currTime time.Time):
-    // unbonding validators are in ordered queue from oldest to newest
-    for all unbondingValidators whose CompleteTime < currTime:
-        validator = GetValidator(unbondingValidator.ValidatorAddr)
-        if validator.DelegatorShares == 0 {
-            RemoveValidator(unbondingValidator)
-        } else {
-            validator.Status = sdk.Unbonded
-            SetValidator(unbondingValidator)
-        }
-    return
-```
+ - The new validator set is taken as the top `params.MaxValidators` number of
+   validators retrieved from the ValidatorsByPower index
+ - The previous validator set is compared with the new validator set 
+   - missing validators begin unbonding
+   - new validator are instantly bonded
 
-## Validator Set Changes
+in all cases any validators leaving or entering the bonded validator set or
+changing balances and staying within the bonded validator set incure an update
+message which is passed back to Tendermint.
 
-The Tendermint validator set may be updated by state transitions that run at
-the end of every block. The Tendermint validator set may be changed by
-validators either being jailed due to inactivity/unexpected behaviour (covered
-in slashing) or changed in validator power. Determining which validator set
-changes must be made occurs during staking transactions (and slashing
-transactions) - during end-block the already accounted changes are applied and
-the changes cleared
+## Queues 
 
-```golang
-EndBlock() ValidatorSetChanges
-    vsc = GetValidTendermintUpdates()
-    ClearTendermintUpdates()
-    return vsc
-```
+### Unbonding Validators
 
-## CompleteUnbonding
+Each block the validator queue is to be checked for mature unbonding
+validators.  For all unbonding validators that have finished their unbonding
+period, the validator.Status is switched from sdk.Unbonding to sdk.Unbonded if
+they still have any delegation left.  Otherwise, the validator object is
+deleted from state.
 
-Complete the unbonding and transfer the coins to the delegate. Realize any
-slashing that occurred during the unbonding period.
+### Unbonding Delegations
 
-```golang
-unbondingQueue(currTime time.Time):
-    // unbondings are in ordered queue from oldest to newest
-    for all unbondings whose CompleteTime < currTime:
-        validator = GetValidator(unbonding.ValidatorAddr)
-        AddCoins(unbonding.DelegatorAddr, unbonding.Balance)
-        removeUnbondingDelegation(unbonding)
-    return
-```
+Complete the unbonding of all mature `UnbondingDelegations.Entries` by
```

Comment:
I'm assuming this refers to the `UnbondingDelegations` queue? Where is that persisted? This paragraph could also be expanded to improve readability.

Reply:
The unbonding queue state documentation has not yet been updated, will be updated as a part of this issue: https://github.com/cosmos/cosmos-sdk/issues/3324

Reply:
updated the text a bit to be a bit more clear, also fixed the tense issue